api/find.go

```diff
@@ -0,0 +1,95 @@
+package api
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"github.com/puppetlabs/wash/activity"
+	"github.com/puppetlabs/wash/api/rql"
+	"github.com/puppetlabs/wash/api/rql/ast"
+	apitypes "github.com/puppetlabs/wash/api/types"
+)
+
+// swagger:route GET /fs/find find listEntries
+//
+// Recursively descends the given path returning it and its children.
+//
+//     Consumes:
+//     - application/json
+//
+//     Produces:
+//     - application/json
+//
+//     Schemes: http
+//
+//     Responses:
+//       200: entryList
+//       400: errorResp
+//       404: errorResp
+//       500: errorResp
+var findHandler = handler{fn: func(w http.ResponseWriter, r *http.Request) *errorResponse {
+	ctx := r.Context()
+	entry, path, errResp := getEntryFromRequest(r)
+	if errResp != nil {
+		return errResp
+	}
+
+	/*
+		// TODO: Include this?
+		if !plugin.ListAction().IsSupportedOn(entry) {
+			return unsupportedActionResponse(path, plugin.ListAction())
+		}
+	*/
+
+	minDepth, hasMinDepth, errResp := getIntParam(r.URL, "mindepth")
+	if errResp != nil {
+		return errResp
+	}
+	maxDepth, hasMaxDepth, errResp := getIntParam(r.URL, "maxdepth")
+	if errResp != nil {
+		return errResp
+	}
+	fullMeta, errResp := getBoolParam(r.URL, "fullmeta")
+	if errResp != nil {
+		return errResp
+	}
+	var rawQuery interface{}
+	if err := json.NewDecoder(r.Body).Decode(&rawQuery); err != nil {
```

Comment:
Hm, starting to think the Boolean primaries should be added back. Then empty query can just set the rawQuery to `true`.

Reply:
Added it back.